File: leetcode/C++/increasing-order-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<TreeNode*> v;
    void traverse(TreeNode* node){
        if(node==nullptr) return;
        traverse(node->left);
        v.push_back(node);
        traverse(node->right);
    }
    TreeNode* increasingBST(TreeNode* root) {
        traverse(root);
        for(int i=0; i<v.size(); ++i){
            v[i]->left=v[i]->right=nullptr;
            if(i>0) v[i-1]->right=v[i];
        }
        return v[0];
    }
};

File: leetcode/Python/increasing-order-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def increasingBST(self, root: TreeNode) -> TreeNode:
        v = list()
        def traverse(node):
            if node is None:
                return
            traverse(node.left)
            v.append(node)
            traverse(node.right)
        traverse(root)
        for i in range(len(v)):
            v[i].left=v[i].right=None
            if i>0:
                v[i-1].right = v[i]
        return v[0]